### PR TITLE
test: Add tests for reset password page

### PR DESCRIPTION
**Pull request by bistaastha**
## Summary by CodeRabbit

- **New Features**
	- Updated the click event handler for the sign-in redirection on the reset password page.
	- Enhanced error handling with toast messages for better user feedback.

- **Bug Fixes**
	- Improved validation error messaging for invalid email input.

- **Documentation**
	- Updated test cases to verify the functionality and state of the reset password component.

- **Refactor**
	- Renamed CSS classes for better organization and adherence to naming conventions.

### CODE REVIEW

File: src/app/auth/reset-password/reset-password.page.spec.ts [hunk 1]
```diff
@@ -1,54 +1,40 @@
...
-    @Input() successContent: string;
-
-    @Input() sendEmailPageState: PageState;
-
-    @Input() isLoading: boolean;
-
-    @Output() sendEmail = new EventEmitter<string>();
-  }
+  let matSnackBar: jasmine.SpyObj<MatSnackBar>;
+  let snackbarProperties: jasmine.SpyObj<SnackbarPropertiesService>;
```
Comment (Chethan-Fyle) [inline src/app/auth/reset-password/reset-password.page.spec.ts:23]: `let snackbarPropertiesService: jasmine.SpyObj<SnackbarPropertiesService>;`

File: src/app/auth/reset-password/reset-password.page.spec.ts [hunk 2]
```diff
@@ -1,54 +1,40 @@
...
+  let activatedRoute: jasmine.SpyObj<ActivatedRoute>;
+  let formBuilder: jasmine.SpyObj<FormBuilder>;
+  let fb: FormBuilder;
 
   beforeEach(waitForAsync(() => {
     const routerSpy = jasmine.createSpyObj('Router', ['navigate']);
-    const locationSpy = jasmine.createSpyObj('Location', ['path']);
     const routerAuthServiceSpy = jasmine.createSpyObj('RouterAuthService', ['sendResetPassword']);
+    const matSnackBarSpy = jasmine.createSpyObj('MatSnackBar', ['openFromComponent']);
+    const snackbarPropertiesSpy = jasmine.createSpyObj('SnackbarPropertiesService', ['setSnackbarProperties']);
```
Comment (Chethan-Fyle) [inline src/app/auth/reset-password/reset-password.page.spec.ts:32]: `snackbarPropertiesServiceSpy`

**Review by Chethan-Fyle — APPROVED** [review]

File: src/app/auth/reset-password/reset-password.page.scss [hunk 3]
```diff
@@ -160,10 +160,10 @@
...
 
-.back-icon {
-  margin-bottom: 16px;
-  width: 28px;
-  height: 28px;
+  &__back-icon {
+    margin-bottom: 16px;
+    width: 28px;
+    height: 28px;
+  }
```
[the remark this answers is not in the document]
Comment (bistaastha) [reply src/app/auth/reset-password/reset-password.page.scss:168]: not needed

File: src/app/auth/reset-password/reset-password.page.ts [hunk 4]
```diff
@@ -62,6 +64,14 @@
...
       this.router.navigate(['/', 'auth', 'disabled']);
     } else {
+      const toastMessageData = {
+        message: 'Something went wrong. Please try after some time.',
+      };
+
+      this.matSnackBar.openFromComponent(ToastMessageComponent, {
+        ...this.snackbarProperties.setSnackbarProperties('success', toastMessageData),
+        panelClass: ['msb-info'],
+      });
```
[the remark this answers is not in the document]
Comment (bistaastha) [reply src/app/auth/reset-password/reset-password.page.ts:74]: Only generic error message will be used in this case

File: src/app/auth/reset-password/reset-password.page.spec.ts [hunk 5]
```diff
@@ -102,23 +175,34 @@
...
 
-      expect(component.currentPageState).toEqual(PageState.failure);
+      component.handleError(error);
+      expect(matSnackBar.openFromComponent).toHaveBeenCalledOnceWith(ToastMessageComponent, {
+        ...props,
+        panelClass: ['msb-info'],
+      });
+      expect(snackbarProperties.setSnackbarProperties).toHaveBeenCalledTimes(1);
     });
   });
```
[the remark this answers is not in the document]
Comment (bistaastha) [reply src/app/auth/reset-password/reset-password.page.spec.ts:202]: not needed

**Pull request merged**